tests/app/main/views/test_suppliers.py

```diff
@@ -1031,13 +1039,16 @@ def test_can_upload_countersigned_agreement_for_already_countersigned_agreement(
         data_api_client.approve_agreement_for_countersignature.assert_not_called()
 
         s3.S3.return_value.save.assert_called_once_with(
```

Comment:
Offtopic but I think we should move away from using `assert_called_once_with` and family - they're too easy to get wrong and be interpreted as a do-nothing mocked call. Making sure an actual assert statement is evaluated by your own code seems to be the only way to make sure.